### PR TITLE
Iss025

### PR DESCRIPTION
**Pull request by g7gpr**

### CODE REVIEW

File: chasemapper/config.py [hunk 1]
```diff
@@ -54,6 +54,9 @@
     'bearing_color': 'black',
     'bearing_custom_color': '#FF0000',
 
+    # History 
+    'reload_last_position': True
```
Comment (darksidelemm) [inline chasemapper/config.py]: Make this False by default... I certainly wouldn't want this on all the time.

File: chasemapper/logread.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                try:
+                  log = read_file("./log_files/" + file)
+                except Exception as e:
+                  logging.debug("Error reading file - maybe in use")
+                for _entry in log:
+                    if _entry['log_type'] == "BALLOON TELEMETRY":
+                      telemetry_found = True
+                if telemetry_found == True:
+                    logging.info("Last balloon telemetry at %s, %s, %.5f, %.5f" % (_entry['callsign'], _entry['time'], _entry['lat'],_entry['lon']))
+                    _entry['time_dt'] = datetime.fromisoformat(_entry.pop('time'))
```
Comment (darksidelemm) [inline chasemapper/logread.py]: Use the dateutil parser function here, it's a bit more tolerant of formatting differences, and results in a time which is timezone aware.  
```
from dateutil.parser import parse
parse(_entry.pop('time'))
```
par

File: horusmapper.cfg.example [hunk 3]
```diff
@@ -242,7 +242,9 @@ unitselection = metric
 # This is the threshold for switching from miles to feet, set in metres.
 switch_miles_feet = 400
 
+[history]
 
-
+# Enable load of last position from log files (True/False)
+reload_last_postion = True
```
Comment (darksidelemm) [inline horusmapper.cfg.example]: Again, make this default to False.

File: chasemapper/config.py (hunk 1 above)
Comment (g7gpr) [reply chasemapper/config.py]: Done

File: horusmapper.cfg.example (hunk 3 above)
Comment (g7gpr) [reply horusmapper.cfg.example]: Done

File: chasemapper/logread.py (hunk 2 above)
Comment (g7gpr) [reply chasemapper/logread.py]: Done

**Comment by darksidelemm** [comment]
Looking good! I'll give it a test sometime today.

**Comment by g7gpr** [comment]
Thanks - it does exactly what I need. The only thing I don't like is that it writes a log file with the data pulled from the log file, but it does at least get the log_time and time correct. I can't think of an elegant way to avoid this.

**Comment by darksidelemm** [comment]
How about adding an optional argument to https://github.com/projecthorus/chasemapper/blob/master/horusmapper.py#L185  e.g. make it `def handle_new_payload_position(data, log_position=True):`

Then at the end of the function, where the position is logged ( https://github.com/projecthorus/chasemapper/blob/master/horusmapper.py#L273 ) make that `if chase_logger and log_position`

Then when you push data in from the log file, you can just set the log_position argument to False. That'll stop the data ending up in the log file.

**Comment by g7gpr** [comment]
Yes, that's was my first thought, but I'm not convinced it is the best way.

**Comment by darksidelemm** [comment]
It might not be the best way, but it's the obvious solution, which is probably good enough for this situation.

**Comment by g7gpr** [comment]
Agreed. I'll implement like that.

**Comment by g7gpr** [comment]
Implemented, but it will need testing this evening.

**Comment by darksidelemm** [comment]
There's a new function in auto_Rx (you may need to run a git pull) where you can 'play back' one of the auto_rx sonde log files.

Go into the radiosonde_auto_rx/auto_rx directory and run:
`python3 -m autorx.emulation log/path_to_log_file.log`

This will play back the log file in 'realtime' out the default UDP port of 55673. The timestamps are replaced with the current time (incrementing as appropriate).

**Comment by g7gpr** [comment]
I'll have a play with that, unfortunately out of time on this for now.

**Comment by g7gpr** [comment]
Failed testing. I think I know what the problem it.

**Comment by g7gpr** [comment]
Tested on live data from S3551348.

**Comment by g7gpr** [comment]
The predictor algorithm is running for the historical data and producing garbage as there is no vertical movement, so I've blocked the predictor running when there is 1 or fewer points in the payload track.

**Comment by g7gpr** [comment]
I think it's done.

**Comment by darksidelemm** [comment]
Will have a test when I get a chance (probably Friday)

**Pull request merged**